### DISCUSSION TITLE
Why is EnableHardwareAccelerationInRdp an opt-in ?!

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/wpf#8280] [Q&A] [answered]
Can someone explain in a short answer why hardware acceleration in RDP has been made an opt-in instead of opt-out?! On the [whats new in .NET 8](https://github.com/dotnet/core/issues/8439#issuecomment-1711374979) its just saying that this new option is so much better versus the old. No disadvantage is mentioned. Is there one? And if not, why its an opt-in then?!

PS: I also tried reading discussion in the [PR](https://github.com/dotnet/wpf/pull/7684) but I couldn't find a clear statement for someone not knowing whats going on in code base. And it seems no documentation exists yet.

**Comment by pchaurasia14** [comment]
Below are the reasons why this is made opt-in instead of opt-out -

1. Changes are at a relatively lower-level (wpfgfx). 
2. Lack of adequate test coverage in this area
3. The change was very close to .NET 8 release cycle, which does not give us enough time to fix any issues (should there be any)

This is explained in the comments [here](https://github.com/dotnet/wpf/pull/7684#issuecomment-1628405496) and [here](https://github.com/dotnet/wpf/pull/7684#issuecomment-1630680002)

**Reply by Symbai** [reply]
Sounds like "we didn't have enough time to fully test it". So with .NET 9 it could become enabled by default?!

**Reply by pchaurasia14** [reply, marked as answer]
Yes, that's the plan.

**Reply by mo-russo** [reply]
@pchaurasia14 Any updte on this? I don't think I've seen any mention of this in the .NET 9 release notes. Could you confirm if it's enabled by default in .NET 9?
